process/sync/metaForkDetector.go

```diff
@@ -40,6 +49,7 @@ func (mfd *metaForkDetector) AddHeader(
 	state process.BlockHeaderState,
 	finalHeaders []data.HeaderHandler,
 	finalHeadersHashes [][]byte,
+	isNotarizedShardStuck bool,
```

Comment:
i think this could be consumed from an interface, not necessarily sent through as a parameter.

Reply:
Yep, I will think about it